### DISCUSSION TITLE
convert gguf weight to hf

### DISCUSSION THREAD

**Discussion by ha-seungwon** [ggml-org/llama.cpp#9617] [Q&A]
Hello

I want to convert llama2-7B gguf weight to hf weight

How can I do that???

**Comment by ngxson** [comment]
There is no straight-forward way to do. One way would be to:
1. Firstly, load the base model from HF
2. Then, load gguf weights into pytorch using `gguf-py`
3. Replace base weights with the loaded weights from gguf
4. Save the model
